sdk/OoyalaSkinSDK/discoveryPanel.js

```diff
@@ -69,6 +81,30 @@ var DiscoveryPanel = React.createClass({
     }
   },
 
+  onStatusPressed: function() {
+    this.setState({showCircularStatus: false});
+  },
+
+  setCounterTime: function(time) {
+    this.setState({
+      currentCounterVal: time,
+      counterLimit: time,
+      showCircularStatus: true,
+    });
+  },
+
+  updateTimer: function(row) {
+    if (this.state.currentCounterVal == 0) {
+      this.onRowSelected(row);
+    } else {
+      var self = this;
+      setTimeout(function() {
```

Comment:
I'm thinking of having a module (maybe residing within the CircularStatus.js) that would send an event to the UI (discoveryPanel.js) that a new second thick in the clock happened. So the UI is only responsible of updating the UI by receiving this event. But I really don't know how hard it would be to implement it.